### ISSUE TITLE
Constant OpenAL errors causing huge I/O writes resulting in poor performance, audio loss & high drive memory usage

### ISSUE THREAD

**Issue by bonenaut7** [anegostudios/VintageStory-Issues#4746]
### Game Version

v1.19.8

### Platform

Windows

### Modded

Vanilla

### SP/MP

Multiplayer

### Description

After playing a long time, audio just disappears and OpenAL errors starts appearing in the console.
I'm writing the issue after the most violent episode happened, with more than 11mb/s IO writes/sec & client log with size over 5 gigabytes.

Another alarming thing that's one of the log lines says `OutOfMemory`, and game process at this point consumed more than 28GB of RAM with 512 chunks view distance, that was lowered to 256 chunks view distance ~2-5 minutes before the app was shut down.

My system at the moment has 64GB of RAM and 64GB of Virtual memory on disks.

### How to reproduce

Unable to find out how to reproduce it, but that could be achieved with playing for a long time and travelling long distances with translocators very often. The most violent episode could happen in the temporal storm, when frequency of OpenAL calls is very high.

### Screenshots

Process Hacker usage stats:
![image](https://github.com/user-attachments/assets/50fe8deb-f806-425e-82ae-c83563234ee1)

Process hacker total stats:
![image](https://github.com/user-attachments/assets/4707993e-985d-4222-8658-d4d129603d47)

Logs:
![image](https://github.com/user-attachments/assets/6f5020aa-ab4e-424e-872f-9049131bd36c)

Frametime at the moment:
![image](https://github.com/user-attachments/assets/af42cec3-d83f-4b9b-a993-d884705fd3c5)

### Logs

Please take a very close look on the next line:
`20.12.2024 11:46:28 [Warning] OpenAL Error during construction of sound game:sounds/creature/drifter-aggro2.ogg: OutOfMemory`

```
20.12.2024 11:46:28 [Warning] OpenAL Error during disposestop of sound game:sounds/creature/drifter-idle2.ogg: InvalidName
...
20.12.2024 11:46:28 [Warning] OpenAL Error during dispose of sound game:sounds/creature/drifter-idle2.ogg: InvalidName
...
20.12.2024 11:46:28 [Warning] OpenAL Error during SetPitch of sound game:sounds/block/quern.ogg: InvalidName
...
20.12.2024 11:46:28 [Warning] OpenAL Error during setvolume(val) of sound game:sounds/block/quern.ogg: InvalidName
...
20.12.2024 11:46:28 [Warning] OpenAL Error during stop of sound game:sounds/environment/wind.ogg: InvalidName
...
20.12.2024 12:29:29 [Warning] OpenAL Error during setposition vec3 of sound game:sounds/environment/waterwaves.ogg: InvalidName
...
20.12.2024 12:35:35 [Warning] OpenAL Error during SetPitchOffset of sound game:sounds/environment/fire.ogg: InvalidName
```

And there was a few lines of NUL symbols, check the screenshots

**Comment by bonenaut7** [comment]
I believe that issue could be a duplicate of #4024, could be closed i guess althrough it could contain another useful info if the issue will be present on 1.20

**Comment by radfast** [comment]
Any clue in the log about what happened at around the time when the issue started - I mean the time when memory use suddenly jumped up to 15GB, as shown in your posted graph?

**Comment by bonenaut7** [comment]
@radfast I'll try to test the game in singleplayer with translocators on the 1.20 rc.5 version, i'll let you know if i'll find anything.

About the 15GB peak - it's not a bug, that's just behavior of the Process Hacker itself, since i've made screenshots pretty quick after starting it and looking on the graps, i'm pretty sure that peak is just Process Hacker asked for the application usage stats from the OS or something, and because it didn't had any before, there is a peak

**Comment by bonenaut7** [comment]
Looks like everything is fine on 1.20 rc.5, i've tried to recreate something similar i've seen on the server (a lot of chiseled stuff, beams, all different), teleporting all over the map again & again, using huge view distance, and yet nothing caused any errors
I was testing the world for an hour, and everything seems fine on the 1.20 rc.5

**Comment by radfast** [comment]
> Looks like everything is fine on 1.20 rc.5, i've tried to recreate something similar i've seen on the server (a lot of chiseled stuff, beams, all different), teleporting all over the map again & again, using huge view distance, and yet nothing caused any errors I was testing the world for an hour, and everything seems fine on the 1.20 rc.5

Excellent to hear, I think this can be closed then as "Resolved" in 1.20.   If you do experience issues of this kind in 1.20, you are welcome to re-open this and post relevant 1.20 logs.
